src/main/com/fulcrologic/fulcro/algorithms/tx_processing/synchronous_tx_processing.cljc

```diff
@@ -83,19 +83,26 @@
 
 (declare run-queue! available-work?)
 
+(defn current-thread-id 
+  "Get the current thread id on the JVM. Returns 0 on JS."
+  []
+  #?(:clj (.getId (Thread/currentThread))
+     :cljs 0))
+
 #?(:clj
    (defmacro in-transaction [app-sym & body]
      `(let [id# (:com.fulcrologic.fulcro.application/id ~app-sym)]
-        (swap! apps-in-tx update id# (fnil inc 0))
+        (swap! apps-in-tx update id# conj (current-thread-id))
```

Comment:
`(fnil conj #{})` I think. Or are you creating a stack? If the latter, then `(fnil conj [])`, which would obviate the need for `vec` conversion in the finally step

Reply:
It needs to be a vec since it's a stack. I'll change to `(fnil conj [])` 👍

Reply:
I just realized that `conj`/`pop` is the actual pair I wanted, so I've just changed to that instead.